Fix build error for cy8ckit_062_ble/cy8cproto_062_4343w boards

npal-cy opened the pull request:
Problem#1: not defined zephyr_console for cy8ckit_062_ble@1.0.0/cy8c6347/m4
https://github.com/zephyrproject-rtos/zephyr/actions/runs/11524969451/job/32695831764?pr=80025
```
- west twister -p cy8ckit_062_ble@1.0.0/cy8c6347/m4 -s tests/drivers/build_all/w1/drivers.w1.build
INFO    - or with west:
INFO    - west build -p -b cy8ckit_062_ble@1.0.0/cy8c6347/m4 tests/drivers/build_all/w1 -T drivers.w1.build
```

Problem#2: arm_irq_zero_latency_levels:
https://github.com/zephyrproject-rtos/zephyr/actions/runs/11524969451/job/32695832470?pr=80025
```
INFO    - The following issues were found (showing the top 10 items):
INFO    - 1) tests/arch/arm/arm_irq_advanced_features/arch.arm.irq_advanced_features on cy8cproto_062_4343w/cy8c624abzi_s2d44 error (Build failure)
INFO    - 2) tests/arch/arm/arm_irq_zero_latency_levels/arch.arm.irq_zero_latency_levels on cy8cproto_062_4343w/cy8c624abzi_s2d44 error (Build failure)
```

npal-cy committed:
cy8ckit_062_ble: board: moved uart-6 to common dts for m4

Moved uatr6 to common dts for m4

- cm0 uses uart-5 in ver 0.0.0 and uart-5 in ver 1.0.0 (for
using of Arduino headers).

- cm4 by default uses uart-2 (for ver 0.0.0 and ver 1.0.0)

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

nordicjm reviewed (changes requested):
You have a revision 1.0.0 board so the board targets must exist

npal-cy commented:
For https://github.com/zephyrproject-rtos/zephyr/actions/runs/11524969451/job/32695832470?pr=80025

"Invalid interrupt priority. Values must not exceed IRQ_PRIO_LOWEST".

Default GPIO interrupt priority is 6, in arch.arm.irq_zero_latency_levels test with CONFIG_ZERO_LATENCY_LEVELS=2,  IRQ_PRIO_LOWEST will be 4. 

Options we have here:
1. add board overlay files in tests/arch/arm/arm_irq_zero_latency_levels, where change DPIO priority for used GPIO port.

2. rework default interrutp priority from 6 to defined in IFX_DEFAULT_IRQ_PRIORITY/override.dtsi similar to how Nordic uses.
-- add override.dtsi in tests/arch/arm/arm_irq_zero_latency_levels (dts\arm\infineon)

@nordicjm any suggestion how to handle this in proper way?

npal-cy committed:
Infineon: board: remove CONFIG_GPIO from defconfigs

Remove CONFIG_GPIO from defconfigs for Infineon boards.
Applications, drivers will enable GPIO if need.

Added 'select GPIO' from spi/Kconfig.ifx_cat1
Added 'select GPIO' from wifi/infineon/Kconfig.airoc

Signed-off-by: Nazar Palamar <nazar.palamar@infineon.com>

npal-cy commented:
Hi @nordicjm, i updated title/description and provided fix for problem#1/problem#2. 

Those changes should unblock https://github.com/zephyrproject-rtos/zephyr/pull/80025#issuecomment-2475346828

mmahadevan108 commented on boards/cypress/cy8ckit_062_ble/cy8ckit_062_ble_cy8c6347_m4.dts:
Please fix the typo in the commit message `uatr6` to `uart6` and `uatr5` to `uart5`. Do you need to add this to `cy8ckit_062_ble_cy8c6347_m0.dts`

npal-cy replied:
Fixed commit message. we do not need update for cy8ckit_062_ble_cy8c6347_m0.dts: cm0 uses uart-5 in ver 0.0.0 and uart-5 in ver 1.0.0 (for using of Arduino headers). cm4 by default uses uart-2 (for ver 0.0.0 and ver 1.0.0)

mmahadevan108 commented on boards/infineon/cy8cproto_062_4343w/cy8cproto_062_4343w_defconfig:
I am a bit hesitant to take this commit change this late in the release. Also this does not seem to be addressing the CI failure. 
Can you split this to a separate PR?

npal-cy replied:
This PR fixes CI failures. The updates in [Infineon: board: remove CONFIG_GPIO from defconfigs](https://github.com/zephyrproject-rtos/zephyr/pull/81377/commits/9dc04c1e9e963033d590c16a403be1a6f9f5a7de) fix arm_irq_zero_latency_levels failures (https://github.com/zephyrproject-rtos/zephyr/actions/runs/11524969451/job/32695832470?pr=80025) because GPIO (with interrupt priority 6) will be disabled in arm_irq_zero_latency_levels. GPIO is actually not required by the arm_irq_zero_latency_levels test.
